src/NHibernate/Collection/Generic/SetHelpers/SetSnapShot.cs

```diff
@@ -1,7 +1,7 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
-#if NETCOREAPP2_0
+#if NETCOREAPP2_0 || NETCOREAPP2_1
```

Comment:
Let's do `NETCOREAPP2_0_OR_GREATER` to minimize changes to master.

Reply:
This requires a higher SDK. According to the [documentation](https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-a-target-framework), a .net 5 SDK is required for these `_OR_GREATER` symbols to be supported.
It may be an issue for the async generator.

Reply:
I know that it requires 5.4 sdk, but I think we can register them manually. This would be better than resolve tens of conflicts.

Reply:
ok